ncm-network/src/main/perl/nmstate.pm

```diff
@@ -0,0 +1,720 @@
+        my $searchpath;
+        push @$searchpath, $net->{domainname};
+        my $dnsservers = $net->{nameserver};
+        $nm_dns_config->{'dns-resolver'}->{config}->{search} = $searchpath;
+        $nm_dns_config->{'dns-resolver'}->{config}->{server} = $dnsservers;
+    }
+    return $nm_dns_config
+}
+
+# enable NetworkManager service
+# without enabled NetworkManager, this component is pointless
+#
+sub enable_network_service
+{
+    my ($self) = @_;
+    # vendor preset anyway
+    return $self->runrun([qw(systemctl enable NetworkManager)]);
+}
+
+# keep nmstate service disbaled (vendor preset anyway), we will apply config ncm component.
+# nmstate service applies all files found in /etc/nmstate and changes to .applied, which will keep changing if component is managing the .yml file.
+# we don't need this.
+#
+sub disable_nmstate_service
+{
+    my ($self) = @_;
+    # vendor preset anyway
+    return $self->runrun([qw(systemctl disable nmstate)]);
+}
+
+# check to see if we have active connection for interface we manage.
+# this allow ability to start a connection again if last config run failed to nmstate apply.
+sub is_active_interface
+{
+    my ($self, $ifacename) = @_;
+    my $output = $self->runrun([$NMCLI_CMD, "-t", "-f", "name,device", "conn", "show", "--active"]);
+    # outoput returned by nmclie -t is colon seperated
+    # i.e eth0:eth0
+    my @existing_conn = split('\n', $output);
+    my $found = 0;
+    foreach my $conn_name (@existing_conn) {
+        my ($name, $dev) = split(':', $conn_name);
+        # trim
+        if ("$dev" eq "$ifacename") {
+            # ncm-network will set connection same as interface name, if this doesn't match,
+            # it means this connection existed before nmstate did its first apply.
+            # doesn't break anything as nmstate resuses the conn, but worth a warning to highlight it?
+            if ("$name" ne "$ifacename"){
+                $self->warn("connection name '$name' doesn't match $ifacename for device $dev, possible connection reuse occured")
+            }
+            $found = 1;
+            return $found ;
+        };
+    }
+    return $found;
+}
+
+# check for existing connections, will clear the default connections created by 'NM with Wired connecton x'
+# good to have.
+sub clear_default_nm_connections
+{
+    my ($self) = @_;
+    # NM creates auto connections with Wired connection x
+    # Delete all connections with name 'Wired connection', everything ncm-network creates will have connection name set to interface name.
+    my $output = $self->runrun([$NMCLI_CMD, "-t", "-f", "name", "conn"]);
+    my @existing_conn = split('\n', $output);
+    my %current_conn;
+    foreach  my $conn_name  (@existing_conn) {
+        $conn_name =~ s/\s+$//;
+        if ($conn_name =~ /Wired connection/){
+            $self->verbose("Clearing default connections created automatically by NetworkManager [ $conn_name ]");
```

Comment:
make sure to add the extra rpm in kickstart and packages to make sure this doesn't happen in the first place